stan/math/fwd/scal/fun/log_mix.hpp

```diff
@@ -177,8 +180,10 @@ inline fvar<T> log_mix(double theta, const fvar<T>& lambda1,
   }
 }
 
-template <typename T>
-inline fvar<T> log_mix(const fvar<T>& theta, double lambda1, double lambda2) {
+template <typename T, typename P1, typename P2,
+          typename = std::enable_if_t<std::is_arithmetic<P1>::value>,
+          typename = std::enable_if_t<std::is_arithmetic<P2>::value>>
+inline fvar<T> log_mix(const fvar<T>& theta, P1 lambda1, P2 lambda2) {
```

Comment:
Could make these ala

```cpp
template <typename Fvar, typename P1, typename P2,
          require_all_arithmetic_t<P1, P2>..., require_fvar_t<Fvar>...>
inline auto log_mix(Fvar&& theta, P1 lambda1, P2 lambda2) {
```

The `...` is a neat trick that tries to do parameter pack expansion which causes the alias to go off

Reply:
Agree with Steve here to simplify this as well as the other two log_mix functions in this file.

Reply:
Done.